### PR TITLE
Add the foundation for automated unit testing using the Google Test framework

### PR DESCRIPTION
**Pull request by Kardbord**
Lots of changes in this one due to a pretty big reorganization, as well as the fact that my new neovim setup feels very strongly about the way things should be formatted. 😅 Still working out the kinks there.

Primarily, the purpose of these updates is to get us set up to write unit tests for the parser so we don't have to eyeball the output of `spicy-driver` manually each time we make a change. We can also set up a GitHub action to run the tests automatically on each PR and after each merge.

Changes:
- Consolidated gtest, test data, and btest all under the testing directory
- Added CMake logic to
  - generate C++ headers/sources from spicy, and link our tests against them
  - generate C++ headers containing embedded binary data from our .binpb test data (I may have gotten carried away here, but it beats hard coding relative paths into the unit tests)
- Added a bare-bones unit test

Leaving this in a draft state for now, as it is blocked by https://github.com/zeek/spicy/issues/1752.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,18 +1,26 @@
...
+
+option(SPICY_PROTOBUF_TEST "Include unit tests in the build" OFF)
 
 list(PREPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake")
 find_package(SpicyPlugin REQUIRED)
 
 # Set minimum versions that this plugin needs. Make sure to use "x.y.z" format.
-# spicy_require_version("1.2.0")
-# spicy_plugin_require_version("0.99.0")
+# spicy_require_version("1.2.0") spicy_plugin_require_version("0.99.0")
```
Comment (Kardbord) [inline CMakeLists.txt]:
```suggestion
# spicy_require_version("1.2.0")
# spicy_plugin_require_version("0.99.0")
```

File: cmake/FindSpicyPlugin.cmake
Comment (Kardbord) [inline cmake/FindSpicyPlugin.cmake:1]: I need to go through this file and fix the formatting. My neovim config really screwed up some of the comments.

I'm not sure there are even any non-whitespace changes in here. I may just be able to revert it.

File: tools/build.sh [hunk 2]
```diff
@@ -78,26 +79,10 @@ function build {
...
-    for binpb in $(find ../test-data/ -name '*.binpb'); do
-      cmd="spicy-driver ./*.hlto -f ${binpb}"
-      echo "============================================="
-      echo "Running ${cmd}"
-      echo "============================================="
-      eval "${cmd}"
-    done
-    #pushd "../testing/" >/dev/null
-    #make
+    pushd "testing/gtest"
```
Comment (Kardbord) [inline tools/build.sh]: Missing corresponding `popd`

**Comment by Kardbord** [comment]
Still need to add unit tests for i32 and i64 wire types.

**Review by TnekRex — APPROVED** [review]
Pulled down code base.  Verified unit tests compile, run and all tests pass.  I'll take a look at the implementation details as time permits - maybe take a shot at implementing the unit tests for I32 and I64 wire types as an exercise to get my head around gtest.

**Pull request merged**